src/foam/core/Action.js

```diff
@@ -134,59 +134,58 @@ foam.CLASS({
   ],
 
   methods: [
-    function andSlots(a, b) {
-      return this.ExpressionSlot.create({
-        args: [ a, b ],
-        code: function(a, b) {
-          return a && b;
-        }
-      });
-    },
 
-    function andSlotAndPromise(slot, promise) {
-      return this.andSlots(slot, this.PromiseSlot.create({
-        promise: promise
-      }));
+    function hasPermissions_(x, property) {
+      if ( ! this.permissionRequired || ! x.auth ) return Promise.resolve(true);
+      return Promise.all(this.permissionConfig[property].map(p => {
+        return x.auth.check(null, p);
+      })).then(arr => arr.every(b => b));
     },
 
-    function checkPermission(x, permission) {
-      if ( ! this.permissionConfig || ! x.auth ) return Promise.resolve(true);
-      return x.auth.check(null, permission);
-    },
-
-    function isEnabledFor(data) {
-      return this.isEnabled
-        ? foam.Function.withArgs(this.isEnabled, data)
-        : true;
-    },
-
-    function createIsEnabled$(data$) {
-      return this.createSlot_(this.isEnabled, 'enabled', data$);
+    function createPermissionlessCheckFor_(data$, property) {
```

Comment:
Naming the second argument 'property' is confusing because I expected it would be a Property. Maybe naming something like 'f' or 'fn' would be clearer. And then calling like:
this.createPermissionlessCheckFor_(data$, this.isEnabled)
would be simpler and clearer.